pkg/ebpf/uprobes/attacher.go

```diff
@@ -629,7 +633,9 @@ func (ua *UprobeAttacher) shouldLogRegistryError(err error) bool {
 func (ua *UprobeAttacher) handleProcessStart(pid uint32) {
 	err := ua.AttachPIDWithOptions(pid, false) // Do not try to attach to libraries on process start, it hasn't loaded them yet
 	if ua.shouldLogRegistryError(err) {
-		log.Warnf("could not attach to process %d: %v", pid, err)
+		if ua.attachLimiter.ShouldLog() {
```

Comment:
Should we be checking `attachLimiter` inside of `shouldLogRegistryError` instead?